### PR TITLE
Feature/cit 442 refactor system test cases

### PR DESCRIPTION
**Pull request by abelikt**
## Proposed changes

An updated approach to categorise our current tests into sub-folders.
This PR will replace https://github.com/thin-edge/thin-edge.io/pull/432

This PR only moves tests into other directories, renames tests some and repairs side effects.
No other changes are done.

- Avoided Kebab case
- Not more than one sub-directory
- Meaningful directory structure (hopefully)
    - > see Readme.md 

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue

https://cumulocity.atlassian.net/browse/CIT-442

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [ ] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by PradeepKiruvale — APPROVED** [review]
LGTM

File: tests/PySys/software_management_end_to_end/sm_apt_install_remove_multiple_mixed/run.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 import subprocess
 import sys
 
-sys.path.append("software-management-end-to-end")
+sys.path.append("software_management_end_to_end")
```
Comment (albinsuresh) [inline tests/PySys/software_management_end_to_end/sm_apt_install_remove_multiple_mixed/run.py:18]: You could avoid this in this file and other similar tests by just adding `<pythonpath value="${testRootDir}/software_management_end_to_end" />` to `pysysproject.xml` file just like it's done for other environments.

**Review by albinsuresh — APPROVED** [review]
This is a great improvement and I have no strong objections to merging this PR as-is.

But, I personally have preference for a slightly more hierarchical directory structure, organised at a feature level. For example, everything software management related like the plugin tests, end-to-end tests etc going into a single directory so that I can run all software management tests at once by executing `psysys run` from that single parent directory.

File: tests/PySys/software_management_end_to_end/sm_apt_install_remove_multiple_mixed/run.py (hunk 1 above)
Comment (abelikt) [reply tests/PySys/software_management_end_to_end/sm_apt_install_remove_multiple_mixed/run.py]: Good hint I did that for the tests that use the line above.

**Comment by abelikt** [comment]
Hi @albinsuresh, thanks for the comment. My Idea here was that all tests are on a similar level (one subdirectory) to avoid a directory structure that is nested to deeply. I would propose that we add another hierarchy, as you proposed, when we have added more test.

**Pull request merged**